From 53b91a3337727af2d10eb2a0af6a366d9986a3b6 Mon Sep 17 00:00:00 2001
From: Alex Belits <abelits@marvell.com>
Date: Wed, 16 Sep 2020 08:38:10 -0700
Subject: [PATCH 0870/1921] task_isolation: Add xen-specific hook

xen_evtchn_do_upcall() should call task_isolation_kernel_enter()
to indicate that isolation is broken and perform synchronization.

Change-Id: I442c8c7aa6c48c7b384f7e41363e42ffaf38828b
Signed-off-by: Alex Belits <abelits@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/36005
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/xen/events/events_base.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/xen/events/events_base.c b/drivers/xen/events/events_base.c
index 8c8b00f7d651..47f810ee3a28 100644
--- a/drivers/xen/events/events_base.c
+++ b/drivers/xen/events/events_base.c
@@ -37,6 +37,7 @@
 #include <linux/cpuhotplug.h>
 #include <linux/atomic.h>
 #include <linux/ktime.h>
+#include <linux/isolation.h>
 
 #ifdef CONFIG_X86
 #include <asm/desc.h>
@@ -1577,6 +1578,8 @@ void xen_evtchn_do_upcall(struct pt_regs *regs)
 {
 	struct pt_regs *old_regs = set_irq_regs(regs);
 
+	task_isolation_kernel_enter();
+
 	irq_enter();
 #ifdef CONFIG_X86
 	inc_irq_stat(irq_hv_callback_count);
-- 
2.31.1

